hack/jenkins/verify-jenkins-jobs.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Pass in Jenkins' URL and the location of e2e.sh in $JENKINS and $E2E.
+# This will tell you which tests are in one but not the other.
+
+# It would be great if each test project went through the same file.
+# As it stands, build* go through build.sh, so won't be in e2e.sh.
+# check-links goes through verify-linkcheck.sh
+# test-go goes through gotest-dockerized.sh
+# pull* run on the pull Jenkins instance
+EXCEPTIONS='
+kubernetes-build
+kubernetes-build-1.0
+kubernetes-build-1.1
+kubernetes-check-links
+kubernetes-test-go
+kubernetes-pull-build-test-e2e-gce
+kubernetes-pull-test-unit-integration
+kubernetes-verify-jenkins-jobs
+'
+
+# Scrape e2e.sh for build names.
+E2E_BUILDS=`grep "^  kubernetes-[\.|a-z|0-9|\-]*)$" $E2E | tr -d " )"`
+
+# Use Jenkins' JSON API to find test project names.
+JENKINS_BUILDS=`curl -s $JENKINS/api/json | python -c '
+import sys, json
+try:
+  jobs = json.load(sys.stdin)["jobs"]
+except ValueError:
+  sys.exit(1)
+for job in jobs:
+  print job["name"]
+sys.exit(0)
+'`
+if [ $? -ne 0 ]
+then
+  echo Could not access or read Jenkins JSON API
+  exit 1
+fi
+
+# Searches $2 for each element of $1
```

Comment:
Have a look at the style guide for function comments.  Should include assumed and set global variables, maybe other things too.